TestGRPCAdapter_HandleAnalytics occasionally failing

The TestGRPCAdapter_HandleAnalytics test is occasionally failing in CircleCI. Relevant logs:

```
listening on :[::]:42195
2019-03-01T07:03:53.798973Z	info	adapters	Handler config: "config.Params{ApigeeBase:\"http://127.0.0.1:33581\", CustomerBase:\"http://127.0.0.1:33581\", OrgName:\"org\", EnvName:\"env\", Key:\"key\", Secret:\"secre...\", TempDir:\"/tmp/107380202\", ClientTimeout:&types.Duration{Seconds: 30,\nNanos: 0,\n}, ApiKeyClaim:\"\", Products:(*config.ParamsProductOptions)(0xc000452120), Analytics:(*config.ParamsAnalyticsOptions)(0xc0003f4d20)}"	{"adapter": "org~env"}
2019-03-01T07:03:53.799137Z	info	adapters	starting product manager	{"adapter": "org~env"}
2019-03-01T07:03:53.803805Z	info	adapters	started product manager	{"adapter": "org~env"}
2019-03-01T07:03:53.804518Z	info	adapters	starting quota manager	{"adapter": "org~env"}
2019-03-01T07:03:53.804575Z	info	adapters	started quota manager with 10 workers	{"adapter": "org~env"}
2019-03-01T07:03:53.804857Z	info	adapters	starting analytics manager	{"adapter": "org~env"}
2019-03-01T07:03:53.804938Z	info	adapters	started analytics manager	{"adapter": "org~env"}
2019-03-01T07:03:53.804950Z	info	adapters	created apigee tenant handler	{"adapter": "org~env"}
2019-03-01T07:03:53.805250Z	info	adapters	closing product manager	{"adapter": "org~env"}
2019-03-01T07:03:53.805264Z	info	adapters	closed product manager	{"adapter": "org~env"}
2019-03-01T07:03:53.805269Z	info	adapters	closing quota manager	{"adapter": "org~env"}
2019-03-01T07:03:53.805401Z	info	adapters	closed quota manager	{"adapter": "org~env"}
2019-03-01T07:03:53.805414Z	info	adapters	closing analytics manager: /tmp/107380202/analytics/org/env/temp	{"adapter": "org~env"}
2019-03-01T07:03:53.805849Z	info	adapters	closed analytics manager	{"adapter": "org~env"}
--- FAIL: TestGRPCAdapter_HandleAnalytics (0.01s)
    grpc_adapter_test.go:183: analytics not delivered
```

--- a/adapter/analytics/analytics_test.go
+++ b/adapter/analytics/analytics_test.go
@@ -285,7 +285,7 @@ func TestPushAnalytics(t *testing.T) {
 	tc.SetOrganization("otherorg")
 	tc.SetEnvironment("test")
 	ctx = &auth.Context{Context: tc}
-	if err := m.SendRecords(ctx, wantRecords[t2][0].records); err != nil {
+	if err := m.SendRecords(ctx, sendRecords[t2][0].records); err != nil {
 		t.Errorf("Error on SendRecords(): %s", err)
 	}
 

--- a/adapter/analytics/upload.go
+++ b/adapter/analytics/upload.go
@@ -59,9 +59,14 @@ func (m *manager) upload(tenant string) error {
 		return fmt.Errorf("ls %s: %s", p, err)
 	}
 	var errs error
+	var lastFn string
 	successes := 0
 	for _, fi := range files {
 		fn := path.Join(p, fi.Name())
+		if fn == lastFn {
+			continue
+		}
+		lastFn = fn
 		f, err := os.Open(fn)
 		if err != nil {
 			errs = multierror.Append(errs, err)

